tests/test_events/test_api.py

```diff
@@ -199,15 +199,31 @@ def test_singleton_pattern_working_correctly_for_event_api(self, mock_send: Magi
         )
 
     @patch.object(Boto3Client, "eventbridge", MagicMock())
-    def test_second_send_clears_everything(self) -> None:
+    def test_second_send_does_not_clear_everything(self) -> None:
```

Comment:
That is not exactly what you check here, you test that events are left in lists. This is quite important as someone in the future won't know why we connect clearing with the send logic 😉

Reply:
You skipped the above comment ☝️

Reply:
Not sure what You want me to do here then.

Reply:
I mean `clear` has nothing to do with `send` after your changes and putting it into the test names is misleading.

But I will write more comments about it as the current version of tests is even more messy 🤷